src/client/components/movies/Movies.js

```diff
@@ -1,31 +1,67 @@
 import React from 'react';
-import styles from './Movies.module.css';
-import MovieContainer from './Movie/MovieContainer';
+import { getAllMovies } from '@root/client/store/store';
+import Movie from '@root/client/components/movies/Movie/Movie'
+import './Movies.module.css'
+import Preloader from '../shared/preloader/Preloader';
 
+class Movies extends React.Component {
 
-const Movies = (props) => {
+  constructor() {
+    super();
+    this.isUnmount = false;
+    this.isLoading = true;
+    this.state = {
+      movies: [],
+    }
+  }
+
+  componentDidMount() {
+    this.isLoading = true;
+    getAllMovies().then((mov) => {
+      if (this.isUnmount === false) {
+        this.setState({ movies: [...mov] });
+      }
+    });
+    this.isLoading = false;
+  }
 
-  if (props.movies.length === 0) {
-    return (
-      <div className={styles.emptyMovies}>
-        <span>No films found </span>
-      </div>
-    )
+  componentWillUnmount() {
+    this.isUnmount = true;
   }
-  else {
 
-    let moviesList = props.movies.map((movie)=> {
+  render() {
+
+    if (this.state.movies.length === 0) {
+      if (this.isLoading === true) {
+        return (
+          <div className ="empty-movies">
+            <Preloader />
+          </div>
+        )
+      } else {
+        return (
+          <div className="empty-movies">
+            <span>No films found </span>
+          </div>
+        )
+      }
+    }
+    else {
+
```

Comment:
`const renderMovie = (movie) => (<Movie key={movie.id} movie={movie} />)`

`return (`
`<div className="movies">`
`{this.state.movies.map(renderMovie)}`
`</div>`
`)`

Reply:
Corrected)